roles/nova-control/tasks/main.yml

```diff
@@ -13,41 +13,63 @@
     with_items:
       - nova-cert
 
-  - name: install nova controller services
-    upstart_service: name={{ item }} user=nova
-                     cmd=/usr/local/bin/{{ item }}
-                     config_dirs=/etc/nova
+  - name: install nova controller services (ubuntu)
+    upstart_service:
+      name: "{{ item.name }}"
+      user: "{{ item.user }}"
+      cmd: "{{ item.cmd }}"
+      config_dirs: "{{ item.config_dirs }}"
     with_items:
-      - nova-api
-      - nova-conductor
-      - nova-consoleauth
-      - nova-scheduler
-      - nova-novncproxy
+      - "{{ service_defns[os].nova_api }}"
+      - "{{ service_defns[os].nova_conductor }}"
+      - "{{ service_defns[os].nova_consoleauth }}"
+      - "{{ service_defns[os].nova_scheduler }}"
+      - "{{ service_defns[os].nova_novncproxy }}"
+
+  - name: permit access to nova api (ubuntu)
+    ufw: rule=allow to_port={{ endpoints.nova.port.haproxy_api }} proto=tcp
+    tags: firewall
   when: os == 'ubuntu'
 
 - block:
   - name: remove retired nova controller services
     systemd_service:
       name: "{{ item }}"
-      user: nova
-      cmd: /usr/local/bin/{{ item }}
-      config_dirs: /etc/nova
+      cmd: "{{ (ansible_distribution == 'RedHat') | ternary('/usr/bin/', '/usr/local/bin/') }}{{ item }}"
```

Comment:
Do we want to include these services in the services/meta instead of hardcoding them here?